packages/tools/piw-utils-internal/src/widgets/FilterSelector.tsx

```diff
@@ -1,37 +1,35 @@
-import { createElement, ReactElement, useCallback, useRef, useState } from "react";
-import { DefaultFilterEnum } from "../../typings/DatagridDateFilterProps";
+import { createElement, ReactElement, useCallback, useEffect, useRef, useState } from "react";
 import classNames from "classnames";
-import { useOnClickOutside } from "@mendix/piw-utils-internal";
+import { useOnClickOutside } from "../utils";
 
-const options: Array<{ value: DefaultFilterEnum; label: string }> = [
-    { value: "greater", label: "Greater than" },
-    { value: "greaterEqual", label: "Greater than or equal" },
-    { value: "equal", label: "Equal" },
-    { value: "notEqual", label: "Not equal" },
-    { value: "smaller", label: "Smaller than" },
-    { value: "smallerEqual", label: "Smaller than or equal" }
-];
-
-interface FilterSelectorProps {
+interface FilterSelectorProps<T> {
     ariaLabel?: string;
     name?: string;
-    defaultFilter: DefaultFilterEnum;
-    onChange: (value: DefaultFilterEnum) => void;
+    defaultFilter: T;
+    onChange: (value: T) => void;
+    options: Array<{ value: T; label: string }>;
 }
 
-export function FilterSelector(props: FilterSelectorProps): ReactElement {
+export function FilterSelector<T>(props: FilterSelectorProps<T>): ReactElement {
     const [value, setValue] = useState(props.defaultFilter);
     const [show, setShow] = useState(false);
     const componentRef = useRef<HTMLDivElement>(null);
     useOnClickOutside(componentRef, () => setShow(false));
+
     const onClick = useCallback(
-        (value: DefaultFilterEnum) => {
+        (value: T) => {
             setValue(value);
             props.onChange(value);
             setShow(false);
         },
         [props.onChange]
     );
+
+    useEffect(() => {
+        setValue(props.defaultFilter);
+        props.onChange(props.defaultFilter);
+    }, [props.defaultFilter, props.onChange]);
```

Comment:
Question: Instead resetting everything manually all the way down here, would it be possible to use a `key={props.defaultFilter}` on `FilterSelector` or `FilterComponent` to automatically do so for us?

Reply:
It will not change the initial value of the state..

Reply:
But as far as I understand: if the key changes, then it will re-mount a new one with the updated `props.defaultFilter` value. `const [value, setValue] = useState(props.defaultFilter);` this will be set to the new value again.  The only problem is that it does not trigger the `props.onChange(props.defaultFilter)`, but in `FilterComponent` that also only resets the state (so potentially we could do a key there). I'm just wondering whether it would be possible to avoid this manual work, but there might be too much extra stuff happening besides just resetting the state (which means a key wouldn't cover everything)